automation/WinFormsAutomationApp/AuthenticationHelper.cs

```diff
@@ -50,17 +50,30 @@ internal static class AuthenticationHelper
                     {
                         userIdentifierType = UserIdentifierType.RequiredDisplayableId;
                     }
+
                     string prompt = input.ContainsKey("prompt_behavior") ? input["prompt_behavior"] : null;
-                    result = await ctx.AcquireTokenAsync(input["resource"], input["client_id"], new Uri(input["redirect_uri"]),
-                        GetPlatformParametersInstance(prompt), 
+
+                    // extraQueryParameters can be null, but key must be present to use correct Acquire Token call for claims overload
+                    if (input.ContainsKey("claims") && (input.ContainsKey("extra_query_parameters")))
```

Comment:
you can call claims API without extra QP as well. so, there should not be a check to ensure extra_query_parameters are present for calling claims overload

Reply:
Done.